### PR TITLE
Docs / update docs for workspaces

### PR DESCRIPTION
**Pull request by ChristiaanScheermeijer**
## Description

In the last sprint, I already changed most of the documentation. During this sprint, @langemike reviewed this and made some changes.

> Note: this branch is rebased onto the `feat/accessibility-and-bug-fixes` branch

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,12 +60,6 @@ The JW OTT Webapp is an open-source, dynamically generated video website built a
 
 - Self-hosted JW Players
 
-## Getting started
-
-- Clone this repository
-- Run `yarn` to install dependencies
-- Run `yarn start`
```
Comment (AntonLantukh) [inline README.md:67]: Maybe we could add a script for web package at the top level to run it, like `yarn start:web` ?

File: docs/workspaces.md [hunk 2]
```diff
@@ -105,25 +104,25 @@ bootstrapping the app.
...
 
-Each platform is a standalone application that may use other packages defined in the packages folder as dependency.
+Each platform is a standalone application that may use other modules defined in the packages folder as dependencies.
 
-### Configs
+### Eslint, PostCSS and Stylelint
 
-The configs directory contains packages that are used mainly for configuring common build tools. This ensures these
-configurations are aligned between the different application packages in the `packages/*` and `platforms/*` folder.
+Besides the mentioned packages there are also three utility packages listed in the packages folder. These utility
```
Comment (AntonLantukh) [inline docs/workspaces.md:111]:
```suggestion
Besides the mentioned packages there are also three utility packages listed in the configs folder. These utility
```

File: platforms/web/docs/initialization-file.md [hunk 3]
```diff
@@ -2,11 +2,11 @@
...
 
-For all manual builds (`yarn start` or `yarn build`), the ini file is copied from `platforms/web/ini/.webapp.<mode>.ini` to `platforms/web/build/public/.webapp.ini`, which the application fetches and parses at startup. For production builds, the ini file is stripped of comments and extra whitespace.
-If a file doesn't exist in /ini/.webapp.<mode>.ini, then the template file will first be copied from [`platforms/web/ini/templates`](../platforms/web/ini/templates).
-All .ini files directly inside of `platforms/web/ini` are ignored in git, so you can create your own files locally to run the application with your account parameters without creating conflicts with committed code or leaking your details into source control.
+For all manual builds (`yarn start` or `yarn build`), the ini file is copied from `/ini/.webapp.<mode>.ini` to `build/public/.webapp.ini`, which the application fetches and parses at startup. For production builds, the ini file is stripped of comments and extra whitespace.
+If a file doesn't exist in /ini/.webapp.<mode>.ini, then the template file will first be copied from [`/ini/templates`](/web/ini/templates).
```
Comment (AntonLantukh) [inline platforms/web/docs/initialization-file.md:8]: Looks like this link doesn't work now

File: platforms/web/docs/initialization-file.md [hunk 4]
```diff
@@ -45,12 +45,12 @@ Keep in mind, if the playerLicenseKey ini setting is provided, it will be used e
 
 ### additionalAllowedConfigSources[]
 
-An array of 8-character IDs (entered 1 per line) for app configs in your JWP account (or url paths) that can be used with the [`app-config=<config source>` query param](configuration.md#switching-between-app-configs).
-This may be useful for example if you have a staging or experimental config that you want to be able to test on your site using the [`app-config` query parameter](configuration.md#switching-between-app-configs) without changing the default config that the application loads with for all of your users.
+An array of 8-character IDs (entered 1 per line) for app configs in your JWP account (or url paths) that can be used with the [`app-config=<config source>` query param](web-configuration#switching-between-app-configs).
+This may be useful for example if you have a staging or experimental config that you want to be able to test on your site using the [`app-config` query parameter](web-configuration#switching-between-app-configs) without changing the default config that the application loads with for all of your users.
```
Comment (AntonLantukh) [inline platforms/web/docs/initialization-file.md:49]: Do these links work for you now?

File: platforms/web/docs/developer-guidelines.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  /containers     - UI Containers
+  /hooks          - Custom React hooks
+  /i18n           - Internationalization tools
+  /modules        - Container registry for IoC
+  /services       - Services which connects external data sources to the application
+  /styles         - Global SCSS rules, theming and variables
+  /utils          - Utility functions
+  /App.tsx        - The main React component which renders the app
+  /constants.ts   - Re-usable TypeScript constants 
+  /index.tsx      - The entrypoint
```
Comment (AntonLantukh) [inline platforms/web/docs/developer-guidelines.md:35]: Does it make sense to add `screenMapping.ts` now?

File: packages/common/docs/modularization.md [hunk 6]
```diff
@@ -42,11 +42,11 @@ This is the price we need to pay to remove `inject` decorators from the construc
...
 
-Initialization starts in the [index.tsx](../platforms/web/src/index.tsx) file where we register services. We do it outside of the react component to make services available in different parts of the application.
+Initialization starts in the platform index.tsx file where we register services. We do it outside the React component to make services available in different parts of the application.
 
-The app is loaded in the [useBootstrapApp](../packages/hooks-react/src/useBootstrapApp.ts) hook with the help of the `AppController` which is responsible for retrieving data from the Config and Settings services, initializing the initial state of the application and hitting init methods of the base app's controllers.
+The app is loaded in the [useBootstrapApp](../src/hooks/useBootstrapApp.ts) hook with the help of the `AppController` which is responsible for retrieving data from the Config and Settings services, initializing the initial state of the application and hitting init methods of the base app's controllers.
```
Comment (AntonLantukh) [inline packages/common/docs/modularization.md:49]: Not sure the link works now.

Comment (AntonLantukh) [reply packages/common/docs/modularization.md:49]: The paragraph should probably go to web platform now.

File: README.md (hunk 1 above)
Comment (langemike) [reply README.md:67]: We can add the following to the root package.json:
```
"scripts": {
    "web": "yarn --cwd platforms/web",
}
```

now you are able to perform `yarn web start`and any other script from `platforms/web/package.json` from the root

Comment (ChristiaanScheermeijer) [reply README.md:67]: We can add this shortcut for sure, but I did not add it on purpose. 

All scripts in the root package JSON are (or should be) platform agnostic. Ideally, when using [Yarn v2+](https://yarnpkg.com/features/scripting), we can use arguments like so:

```json
"scripts": {
    "platform": "yarn --cwd platforms/$1 $2",
}
```

Then you essentially have a shortcut for all platforms;

```shell
$ yarn platform web start
$ yarn platform app build
$ yarn platform server lint
...
```

Comment (AntonLantukh) [reply README.md:67]: I think my main concern here is that it is not clear anymore how to try the app if you just look at the README file. So my idea was to make it more explicit how to start with the Web App.

**Comment by ChristiaanScheermeijer** [comment]
@dbudzins was this PR closed accidentally?

**Comment by budzworthy** [comment]
> @dbudzins was this PR closed accidentally?

Ya, not sure what happened there. I don't know how I would even delete a branch on the videodock fork permission-wise.

**Comment by royschut** [comment]
Auto-closed when the target branch got merged perhaps?

**Comment by budzworthy** [comment]
> Auto-closed when the target branch got merged perhaps?

Maybe? Please reopen if necessary!

**Comment by ChristiaanScheermeijer** [comment]
Ah, perhaps it doesn't auto-update when it's cross-repository... Will reopen it, thanks!

**Pull request closed without merging**